helix-term/src/ui/mod.rs

```diff
@@ -117,18 +115,24 @@ pub fn file_picker(root: PathBuf, config: &helix_view::editor::Config) -> FilePi
         // in our picker.
         .filter_entry(|entry| entry.file_name() != ".git");
 
-    let walk_builder = match type_builder.add(
-        "compressed",
-        "*.{zip,gz,bz2,zst,lzo,sz,tgz,tbz2,lz,lz4,lzma,lzo,z,Z,xz,7z,rar,cab}",
-    ) {
-        Err(_) => &walk_builder,
-        _ => {
-            type_builder.negate("all");
-            let excluded_types = type_builder.build().unwrap();
-            walk_builder.types(excluded_types)
-        }
-    };
+    // We want to exclude files that the editor can't handle yet
+    let mut type_builder = TypesBuilder::new();
+
+    type_builder
+        .add(
+            "compressed",
+            "*.{zip,gz,bz2,zst,lzo,sz,tgz,tbz2,lz,lz4,lzma,lzo,z,Z,xz,7z,rar,cab}",
+        )
+        // We don't care if this fails, but we can't leave failures
+        // unhandled, so just convert to Option and ignore.
+        .ok();
```

Comment:
I'd prefer `.expect("Invalid type definition")`. We know it won't panic since it's a constant. But if it changes it should be caught.

Also, this could have used some of the default types: `xz`, `zip`, `tar`, `lzma` etc.

https://github.com/BurntSushi/ripgrep/blob/master/crates/ignore/src/default_types.rs#L261

Reply:
Expect added. I tried refactoring to take advantage of the default types, but a) they were incomplete and b) I couldn't get them to work nicely and c) it became rather verbose.